Kiosk/Bid Fulfillment/RegisterViewController.swift

```diff
@@ -33,36 +33,21 @@ class RegisterViewController: UIViewController {
         super.viewDidLoad()
 
         coordinator.storyboard = self.storyboard!
-        let registerIndex = coordinator.currentIndex.asObservable()
-        let indexIsConfirmed = registerIndex.map { return ($0 == RegistrationIndex.confirmVC.toInt()) }
+        let indexIsConfirmed = coordinator.currentIndex.map { return ($0 == RegistrationIndex.confirmVC.toInt()) }
 
         indexIsConfirmed
             .not()
             .bindTo(confirmButton.rx_hidden)
             .addDisposableTo(rx_disposeBag)
 
-        registerIndex
+        coordinator.currentIndex
             .bindTo(flowView.highlightedIndex)
             .addDisposableTo(rx_disposeBag)
 
         let details = self.fulfillmentNav().bidDetails
         flowView.details = details
         bidDetailsPreviewView.bidDetails = details
 
-        flowView
-            .highlightedIndex
-            .asObservable()
-            .distinctUntilChanged()
-            .subscribe(onNext: { [weak self] (index) in
-                if let _ = self?.fulfillmentNav() {
-                    let registrationIndex = RegistrationIndex.fromInt(index)
-
-                    let nextVC = self?.coordinator.viewControllerForIndex(registrationIndex)
-                    self?.goToViewController(nextVC!)
-                }
-            })
-            .addDisposableTo(rx_disposeBag)
-
```

Comment:
Cool cool, this threw me off for a second, makes sense

Reply:
Yeah, moving to the next VC is handled here instead: https://github.com/artsy/eidolon/pull/678/files#diff-f37ecaa1c56ebb0278e22df551209993R71 So it would get in this loop because calling the `viewControllerForIndex` function in the subscribe block actually _updates_ the index. Like I said, the improved logging in RxSwift helped track this down 👍 Good job @kzaher.

Reply:
Glad it helped :)